cnjfacil/extrator.py

```diff
@@ -35,9 +36,20 @@ def cnjs(self):
         """Obtem a lista de cnjs encontrados no texto"""
         if not self._cnjs:
             self._busca_cnjs()
+        self._valida_ano_do_cnj()
 
         return self._cnjs
 
+    def _valida_ano_do_cnj(self):
+        limite_inferior = 1895  # Ano do primeiro processo ajuizado do Brasil
+        limite_superior = datetime.datetime.utcnow().year
+        cnjs_invalidos = filter(
```

Comment:
Que tal invés de gerarmos uma lista com os CNJs a remover, não invertemos a lógica dessa lambda e já filtramos as válidas?

Assim evitamos esse `for` com `list.remove`. Que é O(n²).

Afinal você percorre todos os inválidos, e para cada invalido você percorre a lista o buscando.

Reply:
Alteração feita. Realmente fez muito mais sentido e menos complexo.
Obrigado pela sugestão.